packages/react/src/time-input/docs/overview.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: Time input
+description: The Time input component allows users to enter a time free-form, and in various formats.
+group: Forms
+storybookPath: /story/Forms-TimeInput--basic
+unreleased: true
+figmaGalleryNodeId: REPLACE_ME
+relatedComponents: ['text-input']
+---
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState({ raw: '' });
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	const invalid = Boolean(value.raw) && !isValidTime(value.raw);
+
+	return (
+		<TimeInput
+			invalid={invalid}
+			label="Time"
+			message="Enter a valid time"
+			onChange={onChange}
+			value={value}
+		/>
+	);
+};
+```
+
+Time input is superset of [Text input](/components/text-input), with additional functionality specific to entering times. It accepts all of Text input’s props, with changes to `onChange` and `value`.
+
+Time input is a controlled component which means consumers of this component need to manage the state of this component by using the `value` and `onChange` props.
+
+The `onChange` handler provides an object with both a `raw` and `formatted` version of the time. `raw` is always in `'HH:mm'` and `formatted` is provided in the `timeFormat` used. You can use this to display the time elsewhere in the UI, where necessary.
+
+<DoHeading />
+
+- use when users are specifically required to enter a specific time.
+
+<DontHeading />
+
+- use when users are required to enter free-form text that isn’t a time.
+
+## Invalid
+
+Use the `invalid` prop to indicate if the user input is invalid. The `isValidTime()` function can be used to help identify whether the input is valid.
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState({ raw: '9:66' });
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	const invalid = Boolean(value.raw) && !isValidTime(value.raw);
+
+	return (
+		<TimeInput
+			invalid={invalid}
+			label="Time"
+			message="Enter a valid time"
+			onChange={onChange}
+			value={value}
+		/>
+	);
+};
+```
+
+## Changing the time format
+
+The Time input component allows users to enter dates in various formats. To input 9:00 pm, a user could input '9pm', '9:00pm', '2100', '21:00', or other minor variations on these formats.
```

Comment:
Yeah, that's a problem with the user-time library. I tried to fix it, but couldn't and figured that it doesn't matter too much since a user will see that's different to wat they may have expected with their wacky typo

Reply:
You could check if the string has a "p" in it and add an "m" after the p before passing it into the user-time library. I did a small test and "9pmm" does get converted to "9:00 pm"

Reply:
Done